### ISSUE TITLE
Need basic help with installation - need more details

### ISSUE THREAD

**Issue by bgmess** [jlesage/docker-crashplan-pro#34]
I am fairly new to Linux and Docker.  I need more instructions that those given in the quick start:

Launch the CrashPlan PRO docker container with the following command:

docker run -d \
    --name=crashplan-pro \
    -p 5800:5800 \
    -p 5900:5900 \
    -v /docker/appdata/crashplan-pro:/config:rw \
    -v $HOME:/storage:ro \
    jlesage/crashplan-pro

Where do I enter the above commands?

Is this all I need to do?

Any tips would be greatly appreciated.  Not sure how to begin.  I have installed Docker on the Synology NAS, but that's as far as I got.

Thanks,
Brian

**Comment by excalibr18** [comment]
Hello Brian,
I'm a Synology user as well and was using Patters' CrashPlan package before migrating to the Docker image.  Since you're new to Linux, I'll assume you haven't used an SSH client like PuTTY.  You'll need to download that before you get started.  You can get it here: [http://www.putty.org/](url) 

Here's how I got jlesage's crashplan pro docker image up and running on my Synology:

**From Synology DSM:**
1. Open Package Center.
2. Stop Patter's CrashPlan package if you are migrating from it.
3. Install the Docker package from the Package Center interface (this you already did).
4. Close Package Center.
5. Open Docker.
6. Click 'Registry' on the list to the left (below 'DSM').
7. In the search bar at the top of the Docker window, enter `jlesage` and click the 'Search' button. You should see `jlesage/crashplan-pro` at the top of the list. Highlight it and click the 'Download' button. Leave the tag as `latest` in the 'Choose Tag' window that pops up and click the 'Select' button.
8. You should now see a '1' pop up to the right of 'Image' below 'Registry'.  Let your Synology complete the download (it's a 499 MB download).
9. Once the download is complete, open Control Panel from the DSM interface.
10. Click on 'Terminal & SNMP' from Control Panel.
11. Click the box next to 'Enable SSH service'.
12. Open PuTTY.

**From PuTTY:**
1. In the 'PuTTY Configuration' window, enter the local IP address (192.168.xxx.xxx) for your Synology where it says 'Host Name (or IP address)'. Then click the 'Open' button, which will open up a Terminal Session.
2. Enter the username and password at the prompt for a user with Admin privileges.
3. Enter the following command to get 'root' access:
`sudo -i` 
Press Enter.
Enter your password again and press Enter.
4. Quick note about PuTTY: if you highlight the code blocks on this site and copy with Ctrl+C, you'll paste in PuTTY with the Right-Click Mouse Button instead of Ctrl+V.
5. You will need to create the `/docker/appdata/crashplan-pro` configuration folder manually by entering the following command:
`mkdir -p /volume1/docker/appdata/crashplan-pro`
Press Enter.
6. Create the Docker container image by entering the following command at a minimum (there are other environment variables you can introduce if you see fit or necessary - just check jlesage's documentation on those specifics) as a single continuous string (no line break) and let it do its thing:
`docker run -d --name=crashplan-pro -e USER_ID=0 -e GROUP_ID=0 -p 5800:5800 -v /volume1/docker/appdata/crashplan-pro:/config:rw -v /volume1/:/volume1:ro jlesage/crashplan-pro`
The reason we are doing this through SSH/PuTTY is because the Synology Docker program won't let you map Volume1.
7. At this point your Docker container has been created. You can now close your PuTTY session window.

**From Synology DSM:**
1. Uncheck the box next to 'Enable SSH service' in the 'Terminal & SNMP' window in the Synology DSM interface.
2. Open Docker.
3. Click 'Container' from the list on the left. You should now see a 'crashplan-pro' entry with it showing as 'Running'.

**From you Web Browser:**
1. Log into the CrashPlan PRO container GUI by pointing your browser to your IP address of the synology at port 5800 (`http:\\192.168.XXX.XXX:5800`)
2. Sign into your CrashPlan account through the web GUI you just accessed.
3. The following assumes you were using Patters' CrashPlan Package initially and you need to 'adopt' the backup.
4. Click the 'Replace Existing' button to start the wizard.
5. Skip 'Step 2 - File Transfer'.
6. Once done with the wizard, go to your 'Device Details' and click 'Manage Files'.  Since we mapped the Synology's 'Volume1' to the Docker Container's 'Volume1', the Docker CrashPlan-Pro Container should automatically see and recognize the files/folders previously backed up by Patters' CrashPlan Package (assuming you were using it).
7. Perform a backup and the system will make sure it is the same by scanning all the files and making sure everything matches.  If done correctly you shouldn't need to re-upload everything to CrashPlan's servers.
8. You can close the browser window and you’re done! As long as the Docker Container is up and running, you’ll be backing up to CrashPlan. Any changes to your backup set like adding new folders to your backup will be done through the web GUI.

I hope this helps you get up and running!

**Comment by jlesage** [comment]
You can also look at Synology's documentation:
https://www.synology.com/en-global/knowledgebase/DSM/help/Docker/docker_container

**Comment by beagleboy1010** [comment]
Thank you for the step by step instructions. I am a new to docker too. Everything installed and all working fine :) cheers

But I have a problem when I do a restore to the original folder, nothing restored? is this something to do with folder structures?
Please help?

many thanks

**Comment by excalibr18** [comment]
I haven't yet had to restore from within the Docker Container, but I'm fairly certain the reason the restore doesn't work is because Volume1 is being mapped as Read Only: `-v /volume1/:/volume1:ro`

If you change the `-v /volume1/:/volume1:ro` portion of the `docker run` command to `-v /volume1/:/volume1:rw` then it should allow the restores.

@jlesage: is that correct? would the user need to remove the docker container and re-create it with the read/write tag?

**Comment by jlesage** [comment]
Yes that's correct @excalibr18. Re-creating the container with the R/W permission for the volume would allow the restore to work properly.

**Comment by beagleboy1010** [comment]
Perfect! Thanks guys
All working properly now

**Comment by Slartybart** [comment]
Is there a way to map other external drives, as most of my Crashplan files are on drives other than the Volume 1 internal disks?  Thank in advance for any help

**Comment by jedinite13** [comment]
I am having problems connecting with the web service. After running the command and going to the webpage I get Code42 cannot connect to its background service. Retry

If I create the container via the UI it works but I can't create the storage volume to map to /volume1 where all my data is contain in different shares. Any guidance would be appreciated.

Also, as a note, I used the default instructions and it's moved all my files on CrashPlan to Deleted. so be careful when setting $HOME:/storage as your location if you already have a backup set. I am not sure if I am going to have to just upload everything again 2.1TB or if it will de-dup and mark the files as not deleted.

**Comment by jlesage** [comment]
@Slartybart, yes it's possible.  You just need to map additional folders to the container (using the `-v` argument).

**Comment by jlesage** [comment]
@jedinite13, did you follow instructions indicated at https://github.com/jlesage/docker-crashplan-pro#taking-over-existing-backup?
Basically, if path to your files (as seen by the container) is different between your old and new installation, you need to re-select your files using the new paths.  Old paths would be marked as "missing", but that's not a problem.  Once you perform the backup, nothing will be re-uploaded because of deduplication.

**Comment by excalibr18** [comment]
@Slartybart The Synology maps USB external drives as `volumeUSB#` where `#` corresponds to a separate physical USB external drive.  If you only have one USB external drive connected, then it'll be mapped as `volumeUSB1` by default.  

If you're looking to mount the entire USB external drive (in a similar way as Volume1 internal disks are mapped), then you can use `-v /volumeUSB1/usbshare/:/usbshare:ro` when setting up the container.

**Comment by excalibr18** [comment]
@jedinite13 The Synology UI won't allow you to map Volume1, which is why you need to do it via the command line.  

As for not being able to connect to the background service, try doing this:

1. Remove the crashplan-pro docker container.  You can do this from the Docker GUI interface on the Synology.
2. Delete all the contents of `/docker/appdata/crashplan-pro/` (including all the sub folders).  You can do this from Synology's FileStation if you're logged in with an admin-level account, otherwise you'll need to do it from the command line logged in as root.  Make sure you don't actually remove the `crashplan-pro/` folder.  If you did you'll need to re-create it.
3. Create the Docker image again starting from step 6 under "From PuTTY".

That has seemed to worked for other users (see [https://github.com/jlesage/docker-crashplan-pro/issues/14](url))

**Comment by excalibr18** [comment]
One other thing to be aware of: if on the Synology you are experiencing the iNotify Max Watch Limit issue, please refer to this solution: [https://github.com/jlesage/docker-crashplan-pro/issues/23](url)

Also be aware that you'll have to repeat setting the max watch limit in `/etc.defaults/sysctl.conf` each time the DSM downloads and applies an update.

**Comment by aagrawala** [comment]
@excalibr18 & @jlesage Totally AWESOME, folks! Thank you so much! Just followed the instructions instead of fussing with the client app on my MAC.

I have the following questions... Kind of dumb ones I think ;-)

1. We don't need the MAC or local client anymore, correct?

2. Can we delete the java JRE packages from the 'public' folder that we used to download per Patter's procedure?

3. Can we uninstall the CrashPlan Home (Green) package from Synology DSM?

4. With this Docker Package solution, looks like we don't need to install the CrashPlan Pro package from the Package Center in Synology DSM, correct?

Once again, thank you!
Anil

**Comment by excalibr18** [comment]
> 1. We don't need the MAC or local client anymore, correct?

Correct.  I had the local client installed on my Win10 machine and after going with jlesage's docker solution, I no longer need to use that machine. I just need any machine on my local network with a web browser to access the docker container's web GUI.

> 2. Can we delete the java JRE packages from the 'public' folder that we used to download per Patter's procedure?

I would assume this is correct, however when used Patter's package I always opted to use the system Java, not Patter's internal one.

> 3. Can we uninstall the CrashPlan Home (Green) package from Synology DSM?

Assuming you upgraded you CrashPlan subscription to the Pro/Small Business plan, and jlesage's docker solution is confirmed to be working for you (i.e. you have adopted your backup set and it is working correctly), then I don't see a need to keep the CrashPlan Home (Green) package.

> 4. With this Docker Package solution, looks like we don't need to install the CrashPlan Pro package from the Package Center in Synology DSM, correct?

Correct.  With this Docker solution, you don't need the packages from Patter's.

**Comment by aagrawala** [comment]
@excalibr18 Thanks for responding so quickly to my questions.

Another Question: My migration worked fine per the above instructions and the backup is running now looking at the web GUI. However, I cannot seem to browse the files and/or folders that have been backed up using "Manage Files". I can see that volume1 is listed, but browsing under it can't seem to find my file structure that's set up for backup. Any tips?

Thank you in advance!
Anil

**Comment by excalibr18** [comment]
To make sure I understand you correctly, are you saying that when you click "Manage Files" in the GUI, there's nothing listed?  Or are there folders listed, but you just can't find the folders you are backing up?

**Comment by aagrawala** [comment]
Folders listed but can't find my folder structure from my Synology Home directory that I have selected to be backed up. See the attached snapshot showing what's under volume1
<img width="150" alt="screen shot 2018-02-05 at 12 27 39 am" src="https://user-images.githubusercontent.com/31644943/35794685-8946a5b8-0a0b-11e8-9123-fbe861ecbeff.png">

**Comment by excalibr18** [comment]
What were the folder paths for your backup when you were using Patters' package?

**Comment by aagrawala** [comment]
Something like this as shown in the attached from the old client app on my computer saved a few years ago 
<img width="502" alt="screen shot 2018-02-05 at 12 46 49 am" src="https://user-images.githubusercontent.com/31644943/35795527-46b99522-0a0e-11e8-98f3-1c1c6571796e.png">

**Comment by aagrawala** [comment]
Web GUI shows that the backup is running and the files that are new and hadn't been backed up for the last 25 days (yes, I hadn't done this upgrade for so long after the backup had stopped) are being backed up (can't see the files that are being backed up themselves... Just from the size of the backup remaining

**Comment by excalibr18** [comment]
OK this helps. In this case using the screenshot from the old client app, when you scroll down in the "Manage Files" window you should see a "photo" folder in the list. If you click on that "photo" folder, then you should see the various Year folders you had selected in the old client to back up. Each of the Year folders should have a check mark next to it.

**Comment by aagrawala** [comment]
Ah, LoL, I didn't notice the scroll bar to the right side in the GUI and thought those were the only folders what I sent you :-) I do see all the other folders when I scroll down.

Thanks a million for your help. And, sorry for bugging you for this silly thing and taking up your time.

Best,
Anil

**Comment by excalibr18** [comment]
No worries.  Glad it's all sorted out and working for you!  The new GUI for CrashPlan Version 6 is less intuitive than the one before it unfortunately.

**Comment by lenny81** [comment]
Thanks for the installation steps. I've just transitioned to the docker from Patter's solution. Everything went well and the container is running CP pro, however I can't access the web gui. I've followed this step:
"Log into the CrashPlan PRO container GUI by pointing your browser to your IP address of the synology at port 5800 (http://192.168.XXX.XXX:5800)"   and replacing the XXX.XXX with my synology server address. 
I get a 'Can't connect to a server' message. Am I missing something obvious?

Any suggestions would be appreciated.

**Comment by jlesage** [comment]
Which parameters did you give to the `docker run` command?  Did you mapped the port (`-p 5800:5800`)?

**Comment by excalibr18** [comment]
@lenny81: Are you accessing the web GUI of the Docker on the same LOCAL network as the Synology?

**Comment by lenny81** [comment]
I'm trying to access it on the same network as the synology. 

jlesage: No idea. I'm pretty clueless with all this. I can't remember which guide I used to set it up or how I check if I mapped the port... how do I check this?

**Comment by jlesage** [comment]
Using the Synology UI, you can access the container's advanced settings and then there is a "Port Settings" tab where you can see the port mappings.

**Comment by lenny81** [comment]
Thanks for the reply. Under details -> Port settings I can see port settings for ports 4243 and 4242 but not 5800 which I means it’s not mapped. Is there a way I can map them now or will I need to reload the container?

**Comment by excalibr18** [comment]
You can either delete and re-create the container with the correct port mapping from the command line; or stop the container from within the Synology Docker GUI, click the Edit button, go to Port Settings, and change the port to be 5800.

**Comment by lenny81** [comment]
Thanks for the help jlesage and excalibr18, I ended up deleting and reinstalling the container using your guide above excalibr18 and it's all working fine now.

**Comment by halteach** [comment]
so, everything seems okay.  I can't figure out how to force a backup.

**Comment by halteach** [comment]
@excalibr18 Excellent job!  Up and running in less than 30 minutes (It took this long because I didn't follow your instructions in the order they were written!)

Thanks

Hal
refugee from Patters CrashPlan packages for Synology NAS""

**Comment by kmstrs** [comment]
@excalibr18 From the command line I got this: "Error starting userland proxy: listen tcp 0.0.0.0:5000: bind: address already in use."  It starts if I use 6000:6800, but then I can't connect to the web UI using port 6800
I tried changing the port to 5000:5800 from the gui, and then I'm getting "Local port 5000 conflicts with other ports used by other services".  I have no other Docker containers running.
"netstat -an" just shows tcp        0      0 0.0.0.0:5000            0.0.0.0:*               LISTEN  

Anyone else see this?

**Comment by jlesage** [comment]
When you add a port mapping, the first port is the one on the host side and the second one if the one on the container side.  Container side should always be the same (`5800`).  So you could set something like `XXXX:5800`, where `XXXX` is any port not already in use on the host.

**Comment by bigtfromaz** [comment]
Port 5000 is the default port used by the Synology administrative web site.

Tom Hebert
480-788-6601

From my phone, so this message may be more terse than normal

________________________________

**Comment by bgmess** [comment]
When following the directions, I get an error after executing:

docker run -d --name=crashplan-pro -e USER_ID=0 -e GROUP_ID=0 -p 5800:5800 -v /volume1/docker/appdata/crashplan-pro:/config:rw -v /volume1/:/volume1:ro jlesage/crashplan-pro

 

I get the error:

 

:~$ docker run -d --name=crashplan-pro -e USER_ID=0 -e GROUP_ID=0 -p 5800:5800 -v /volume1/docker/appdata/crashplan-pro:/config:rw -v /volume1/:/volume1:ro jlesage/crashplan-pro

docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post  <http://%2fvar%2frun%2fdocker.sock/v1.29/containers/create?name=crashplan-pro:> http://%2Fvar%2Frun%2Fdocker.sock/v1.29/containers/create?name=crashplan-pro: dial unix /var/run/docker.sock: connect: permission denied.

See 'docker run --help'.

 

 

What do I need to do to give it permission?

 

Thanks,

 

Brian

**Comment by excalibr18** [comment]
Did you type `sudo -i` before you ran the docker command?

**Comment by bgmess** [comment]
Great catch…I typed sudo -1, not i.

 

I’ll try that again.

 

Brian

**Comment by bgmess** [comment]
That worked…thanks.

 

How do I choose which files on my Nas I want to backup?  I seem to be missing that part.

 

Regards,

 

Brian

**Comment by excalibr18** [comment]
You can do that from within the CrashPlan GUI. Log into it via your website browser and look for a button called "Manage Files". From there it will show a directory listing from which you can select from using check boxes.

**Comment by bgmess** [comment]
I can’t thank you enough.  Your instructions were perfectly clear, and the process worked flawlessly.

 

Thanks!

 

Brian

**Comment by beBerlin** [comment]
Thank you for the article. The description also helped me a lot. Unfortunately I still have a problem.
I got to point 6.
From point 7 I have probelem. Although I see that the "CrashPlan web GUI interface" but I can not log in.
See photo in the link:
Can anybody help me?
https://www.dropbox.com/s/oqt0mgf3wze9i8z/Ohne%20Titel.jpeg?dl=0

**Comment by beBerlin** [comment]
Thanks for the article. The description also helped me a lot. Unfortunately, I still have a problem.

I see the "CrashPlan Web GUI Interface" but I can not log in.
See photo in the link:
Can someone help me?
https://www.dropbox.com/s/oqt0mgf3wze9i8z/Ohne%20Titel.jpeg?dl=0

**Comment by excalibr18** [comment]
Try following the steps in the Readme section:
https://github.com/jlesage/docker-crashplan-pro/blob/master/README.md#device-status-is-waiting-for-connection

**Comment by jlesage** [comment]
Also, make sure you are running the latest version of the container image.

**Comment by dmase2** [comment]
I'm having a problem connecting to the crashplan-pro servers.  I get the same message as beBerlin above.  I've tried clearing the cache and deleting the appdata directory per other suggestions.  I am able to ping the crashplan servers from the NAS.  Any other ideas?

**Comment by jlesage** [comment]
Are running the latest version if the image?

**Comment by dmase2** [comment]
Yes.

**Comment by excalibr18** [comment]
@beBerlin & @dmase2:
1) Can you confirm you are able to log into CrashPlan's web interface (https://web-eam-msp.crashplanpro.com/login/#/login)?

2) Have you tried completely removing the container and deleting everything within the `/docker/appdata/crashplan-pro` folder on the Synology and starting over?

**Comment by dmase2** [comment]
Yes, I have a windows computer that backs up with no problems from the same network.  I just confirmed again that I can access the web site with no issues.

I have removed the container and deleted the docker/appdata/crashplan-pro directory with no success.  

Just to confirm, I'm on version 6.7.0.4503.

This is my startup script for the container.  I'm running it from an SSH connection.

docker run -d --name=crashplan-pro1 -e USER_ID=0 -e GROUP_ID=0 -e CRASHPLAN_SRV_MAX_MEM=3072M -p 5800:5800 -v /volume1/docker/appdata/crashplan-pro:/config:rw -v /volume1/:/volume1:ro jlesage/crashplan-pro

I notice that there is a green unlock icon to the right of the Crashplan for Small Business logo in the upper left is that normal?

**Comment by excalibr18** [comment]
@dmase2:
So when you get to the part of the guide that says:

> From your Web Browser:
> 
> 1. Log into the CrashPlan PRO container GUI by pointing your browser to your IP address of the synology at port 5800 (http:\\192.168.XXX.XXX:5800)
> 2. Sign into your CrashPlan account through the web GUI you just accessed.

Are you saying that your the CrashPlan PRO container is not accepting your CrashPlan login credentials?

You mention you have a Windows computer that backs up with no problems.  Are you paying for multiple devices on your CrashPlan account?  I know they switched to a per-device payment structure when they dropped CrashPlan Home.

As for the green padlock, yes it's normal:
![image](https://user-images.githubusercontent.com/34701645/37442755-46fc6cba-27c5-11e8-9985-c1de39c10a6e.png)

**Comment by dmase2** [comment]
when i address the container via http://192.xxx.xxx.xxx/5800 I get the first part of the log in screen for crashplan pro where I can enter email.  When I enter the email address the I get the following.
![loginfail](https://user-images.githubusercontent.com/17119442/37445833-588fadea-27d7-11e8-8cf6-99f3bed8eadc.PNG)

When I log in from my windows machine, after entering the email address I get a new dialogue box for entering my password.  

The following is the icon I referred to earlier, note it is in "unlock" position.

![unlockedicon](https://user-images.githubusercontent.com/17119442/37445835-5d71d9a0-27d7-11e8-97b0-56b0cee0ecf2.PNG)

Thanks for the help.

**Comment by jlesage** [comment]
Look at `/volume1/docker/appdata/crashplan-pro/log/service.log.0` to have more details about the failure.

The icon in unlock position is normal when you connect over http: it means that the unencrypted connection to the UI succeeded.

**Comment by dmase2** [comment]
From the service.log.0  this appears to be the problem area.  Is the unresolved address on the server side or on my side?

[03.15.18 15:13:54.953 WARN  DefaultGroup .code42.messaging.peer.PeerGroup] PG::DefaultGroup PeerException attempting to connect. RemotePeer-[guid=42, state=CONNECTING]; Session-null e=com.code42.peer.exception.PeerException: IOExcepton opening remote session. guid=42, remoteLocation=TargetAndProxyLocation [DIRECT  -- target=[[42@central.crashplanpro.com:443(server)], pbKRequestable=true]], timeout=30000, e=Unexpected Exception in connect() for remoteAddress=central.crashplanpro.com:443, java.nio.channels.UnresolvedAddressException

**Comment by jlesage** [comment]
On yours.  Run the following commands on your host to see if the address can be resolved in both cases:
```
nslookup central.crashplanpro.com
```
```
docker exec crashplan-pro1 nslookup central.crashplanpro.com
```

**Comment by dmase2** [comment]
![image](https://user-images.githubusercontent.com/17119442/37475410-e98c6b14-282f-11e8-9c90-98ec84a88f5b.png)

dns problem from docker?

**Comment by dmase2** [comment]
@jlesage 
I added "--net=host" to the script and it now seems to be working.  Able to log in and crashplan is beginning the sync.  Thank you for the docker image and the help.  Do you have a shareware contribution method in place?
Doug

**Comment by excalibr18** [comment]
Donation link for jlesage is here:
![image](https://user-images.githubusercontent.com/34701645/37491026-ad0de418-2859-11e8-8ba4-1335cce986f9.png)

**Comment by jlesage** [comment]
@dmase2, if setting the network mode to “host” fixes the issue, then you seems to have an issue with your docker setup.

Maybe this can help?
https://forum.synology.com/enu/viewtopic.php?t=109640
https://www.google.ca/amp/s/amp.reddit.com/r/docker/comments/64tst6/synology_nas_docker_networking_problem/

**Comment by dmase2** [comment]
@jlesage 
Neither link seemed to help.  I have already tried completely unloading docker and re-installing it from scratch.  I'm not sure how to apply the second link's solution to my situation.  It seemed like a temp solution anyway.

**Comment by dmase2** [comment]
I did a bit more research and by creating a user defined bridge was able to get it working without using "--net=host".  There seems to be a DNS issue in the default bridge for my Docker implementation.

**Comment by PNMarkW2** [comment]
I've followed the step and it seems to be running, but I can not get to the container GUI. When I try to go to 192.168.xxx.xxx:5008 I get the message "This site can't be reached. 192.168.xxx.xxx refused to connect". I'm sure I'm missing something simple, yet I verified that I set the port correct, I've checked the firewall setting, clearly, I can get to the NAS since I setup CrashPlan, and run other apps there like Plex.

Any thoughts?

**Comment by jlesage** [comment]
By default, the port is `5800`, not `5008`.  Did you mapped the port to another one?  Which `docker run` command did you use?

**Comment by PNMarkW2** [comment]
As I said, a simple mistake. No, I didn't change it, I was just typing it wrong, several times and in multiple places. :-)

**Comment by jlesage** [comment]
Then more information is needed to determine the problem:  The parameters of the container (the complete `docker run` command you used to create the container or the output of `docker inspect`) and the  container's log (`docker logs`).

**Comment by PNMarkW2** [comment]
Sorry if I wasn't clear, the only problem I had was that I mistyped the port multiple times. Once you pointed out that mistake on my part and I corrected it, I was able to log in. Totally human error on my part.

I do have one question though, I saw this once but can't find it now, what file and setting do I need to modify to adjust the size of the backup? I've been bitten before by CrashPlan only backing up part of a collection because of that.

**Comment by jlesage** [comment]
Not sure to see what you are talking about.  Normally there is no file that need manual modification.

**Comment by PNMarkW2** [comment]
Things seem to be up and running for the first time since the start of Feb, and I thank you for the effort getting this up and running plus sharing it.

However, I now get this message regularly. I've looked for a file or setting that I could tweak, but I haven't found one yet. Any thoughts from the experts? I've found the program crashed a couple times and had to restart it.
![image](https://user-images.githubusercontent.com/4693751/37690383-caf571dc-2c67-11e8-80e0-7e4639315d0d.png)

**Comment by jlesage** [comment]
Here you go:
https://github.com/jlesage/docker-crashplan-pro/blob/master/README.md#crashes--maximum-amount-of-allocated-memory

**Comment by buffoon** [comment]
I've been attempting to get this running for about a week now with no luck.  I hope I'm missing something simple here.  I've deleted the container and the config multiple times and I'm always stuck on the screen with this prompt:
![capture](https://user-images.githubusercontent.com/1099234/37875604-e1d71cec-300f-11e8-9507-987a160de6cc.PNG)

my docker run command is: 

> docker run -d --name=crashplan -p 5800:5800 -e USER_ID=65534 -e GROUP_ID=65534 -v /mnt/config/Docker/crashplan:/config:rw -v /mnt/config/Vault:/storage:ro jlesage/crashplan-pro

The only error in the service.log.0 is that it can't find the .ui_info file (it never deos get created):

> [03.18.18 11:25:02.880 ERROR main         ackup42.service.ui.UIInfoUtility] Unable to find a file to read the address, token, and port out of due to IO, check that the file is readable?, java.io.IOException: Unable to create parent folders for .ui_info file. folder=/.crashplan
STACKTRACE:: java.io.IOException: Unable to create parent folders for .ui_info file. folder=/.crashplan
	at com.backup42.service.ui.UIInfoUtility.setUIConnectionDetails(UIInfoUtility.java:99)
	at com.backup42.service.CPService.writeUIInfoFile(CPService.java:1407)
	at com.backup42.service.CPService.start(CPService.java:668)
	at com.backup42.service.CPService.main(CPService.java:2064)

When I use the nslookup command I get this:

> tmp # nslookup central.crashplanpro.com
> nslookup: can't resolve '(null)': Name does not resolve
> 
> Name:      central.crashplanpro.com
> Address 1: 216.17.8.5 eam-msp.crashplanpro.com
> Address 2: 216.17.8.56 central.crashplanpro.com

service.log.0 file:

[service.log](https://github.com/jlesage/docker-crashplan-pro/files/1845366/service.log)

**Comment by jlesage** [comment]
Looks like a permission issue.  Are you sure that the user you specified can write to `/mnt/config/Docker/crashplan`?

**Comment by buffoon** [comment]
As far as I can tell that user/group has full read/write/execute permissions to that folder.  It is able to write all the other data in that directory.

**Comment by jlesage** [comment]
Can you provide the output of the following commands while the container is running (run on host):
```
docker inspect crashplan
```
```
docker exec crashplan cat /etc/passwd
```
Also, can you try to run the container without mapping `/config`?

**Comment by buffoon** [comment]
Thanks for looking into this.  Here is the output of the inspect and cat commands:
[docker inspect.txt](https://github.com/jlesage/docker-crashplan-pro/files/1853632/docker.inspect.txt)
[cat.txt](https://github.com/jlesage/docker-crashplan-pro/files/1853633/cat.txt)

Without mapping the /config directory, I get the same result.  The service.log.0 without mapping is here:
[service.log](https://github.com/jlesage/docker-crashplan-pro/files/1853636/service.log)

**Comment by jlesage** [comment]
Ok I found a bug that cause the issue you are seeing.  It is caused by the user ID you use, which matches another user in `/etc/passwd`.
I will fix that.  As a workaround, you can edit `/defaults/passwd` in the container and remove the line containing the user `nobody` (id 65534).  Then restart the container.

**Comment by buffoon** [comment]
That did it thanks!

**Comment by beBerlin** [comment]
@excalibr18 
I have reinstalled the container "jlesage / crashplan-pro".
And previously deleted all folders.

I can also log in to https://web-eam-msp.crashplanpro.com/login/#/login?to=%2Flogin%2F%23%2Flogin.

And that's why I do not understand why I can not log in to the web GUI jlesage / crashplan-pro.

Who can help me?

**Comment by jlesage** [comment]
Can you clarify the issue?  Which error are you getting exactly?

**Comment by beBerlin** [comment]
See photo in the link:
https://www.dropbox.com/s/oqt0mgf3wze9i8z/Ohne%20Titel.jpeg?dl=0

**Comment by jlesage** [comment]
Can you provide the `log/service.log.0` log file?  You can find it on the host in the folder you used to map `/config`.

**Comment by beBerlin** [comment]
https://www.dropbox.com/s/w9kswibhqzsl3m1/service.log.0?dl=0
I hope that's the right file?

**Comment by beBerlin** [comment]
I use DS918 +. I did not understand the other instructions. can you put it another way? or help others?

**Comment by jlesage** [comment]
Are you able to access the NAS via SSH?  You first need to enable the SSH service (https://www.synology.com/en-global/knowledgebase/DSM/help/DSM/AdminCenter/system_terminal) and then use PuTTY (https://www.synology.com/en-global/knowledgebase/DSM/tutorial/General/How_to_login_to_DSM_with_root_permission_via_SSH_Telnet).  Once connected, you will be able execute the 2 provided commands.

**Comment by beBerlin** [comment]
I have executed the commands. See image.
https://www.dropbox.com/s/inocihhdmy1neen/Ohne%20Titel%203.jpeg?dl=0

**Comment by bigtfromaz** [comment]
This is bit of a guess but I see that from the DiskStation nslookup received it’s reply from a DNS server it reached using IPv6.  If your container is connected to a bridge network, check to see if the bridge has IPv6 enabled.

**Comment by beBerlin** [comment]
Enclosed the commands:

https://www.dropbox.com/s/yyh2nd4swzy6ei5/Ohne%20Titel%204.jpeg?dl=0

**Comment by jlesage** [comment]
So it seems that DNS resolution has to be done over IPv6 on your network.  And since IPv6 doesn't seem to be enabled for your container, it's not possible for the latter to reach CrashPlan's server.
Which router are you using?

@bigtfromaz, is there a setting to enable IPv6 in Synology?

**Comment by beBerlin** [comment]
FRITZ!Box 6490 Cable

https://www.dropbox.com/s/7q5zjx7hvu4jps2/Foto%2011.04.18%2C%2006%2045%2002.jpg?dl=0

**Comment by bigtfromaz** [comment]
Yes.  When you create a bridge network you can select “Enable IPv6” .  It seems to require an IPv4 address range for all bridge networks.  IPv6 is optional.

It appears that the Synology does not support autoconfiguration.  So using a local subnet would be in order.  Something like  fd01:10:1000:1000::/64 would be in order.  They call it a subnet but in context I think they mean prefix.

It also asks for an IP range which I suspect will be used as a pool of addresses to give to container that are added to the network.  Here is an example of a 32 bit subnet carved out 64-bit prefix.  An example here might be fd01:10:1000:1000:1000:1000::/32 .  You could probably use the prefix for this as well.

The gateway must be set in the same subnet, something like be fd01:10:1000:1000::1  .

I haven’t played with this at all but it should work pretty much the same as an IPv4 bridge network.  Synology support might be helpful in this regard.

With that said, it may be easier to use the host network rather than a bridge.  The container should get a unique address and a correct gateway address when IPv6 autoconfigures.  NAT is not used in IPv6.  It should be secure.

@jlesage do know of a small container I can deploy with just linux/bash in order to do some testing???

Bigtfromaz

**Comment by jlesage** [comment]
@beBerlin, so it seems that your solutions are:
  - Check your router settings to see if there is anything that would block DNS resolution over IPv4.
  - Try to configure a separate bridge with IPv6 enabled.
  - Switch the network mode to "host" instead of "bridge".

@bigtfromaz, you can use an ubuntu baseimage, like `ubuntu:16.04`.  There is probably a baseimage for any distro you want.

**Comment by beBerlin** [comment]
Thank you for the great help. But I do not know how to make the settings.

**Comment by bigtfromaz** [comment]
I will try.  

From you previous post I know that your container is not receiving a public address.  Without one, it can't talk to the internet.

So I first want to confirm that your host machines are being assigned public addresses.  

Let's start there.  On your host machine, please run an ifconfig, or on a PC run an IPCONFIG command.   One of my sandbox machines looks like this:

Ethernet adapter Ethernet 2:

   Connection-specific DNS Suffix  . : myDomain.com
   IPv6 Address. . . . . . . . . . . : 2600:8800:e00:1620:xxxx:xxxxxxxx:xxxx
   Temporary IPv6 Address. . . . . . : 2600:8800:e00:1620:5044:b90f:6577:7bc7
   Temporary IPv6 Address. . . . . . : 2600:8800:e00:1620:9870:1209:40e9:9835
   Link-local IPv6 Address . . . . . : fe80::c19b:5c07:fc1b:6489%12
   IPv4 Address. . . . . . . . . . . : 192.168.x.x
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : fe80::1:1%12
                                       192.168.x.x
You will see addresses that begin with "f", e.g. fe80.  These are private addresses and no traffic to or from those will traverse the internet.  We first need to check that you are receiving a public address, e.g. 2600:8800:e00:1620:xxxx:xxxxxxxx:xxxx .  In this case my ISP has assigned this 64-bit prefix to me.  2600:8800:e00:1620.  It leaves me with 64 bits to use here for local subnets and addresses (about 4 gig).

Please run the ifconfig (Linux) or ipconfig (Windows) command on one or more physical machine and send the results.

**Comment by damaverick82** [comment]
Hello,

I changes my crashplan pro installation on my Synology a few months ago from the Patters version to the lesage docker version.

Everything seems to work fine except for one thing.
The docker instance seems to find a lot more data then possible.
I have a 716+ Synology with 2  3TB disks in raid 1 with about 400GB of data.
The lesage docker client from crashplan says it needs to backup 141TB of data and therefore never finishes. (even when I deselect a lot of data)
The Patters version did see the right amount of data.

I already configured the docker from scratch.
I even configured crashplan completely from scratch by removing everything online

**Comment by jlesage** [comment]
Did you mapped `/storage/` to `/volume1/` ?  If yes, make sure you didn't selected any docker related folders.

**Comment by jlesage** [comment]
More specifically, `/volume1/@docker` should not be selected.

**Comment by damaverick82** [comment]
Ah thanks, that did the trick for me. There was indeed a docker folder mapped. Although in my case is was on volume2 :)

**Comment by beBerlin** [comment]
@bigtfromaz 
I used Mac, what's the command on the Mac?
with "ipconfig" appears: "usage: ipconfig <command> <args>
where <command> is one of waitall, getifaddr, ifcount, getoption, getpacked, getv6packet, set, setverbose "

**Comment by bigtfromaz** [comment]
I not a Mac expert but I suspect the command would be ifconfig:
[]https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man8/ifconfig.8.html

Do check out #79 but also send the listing I requested above.  If you only have IPv6 then, unless you have a static prefix, you will want to use the host network rather than a bridge network.  IPv6 does not use NAT so you can't access the internet with a private address.

**Comment by bigtfromaz** [comment]
Try this.  Open a command window on your host where you are able to run docker commands. _I use putty to open a terminal to my Synology._  If you don't know how to use putty, search for it on Bing or Google.  There is lots of help there.

If you don't know your container name, you can list it with this command:

`sudo docker container ls`


Once you have your container name, enter this command:

`sudo docker exec -i -t jlesage-crashplan-pro1 /bin/sh`

Note that my container name is "jlesage-crashplan-pro1"  .  You should use your container name if it's different than mine.

The docker exec command should give you  a prompt that looks something like this:
`/tmp #`

At this point, your are "in" the container.  Run these commands and send the output.

`ifconfig`
`ip route`

To summarize, use putty to open a command prompt on your Synology.  Log into the Synology using your administrator user id and password.  Determine your container name.  Issue the "sudo docker exec" command to get a command prompt inside the container.  Enter the ifconfig and ip route commands and post the output.  This will tell us about the network configuration inside you container.

**Comment by beBerlin** [comment]
@bigtfromaz 
Thank you so much. Here are the screenshot of the commands.
https://www.dropbox.com/s/sgohojwwmhem6l5/Ohne%20Titel%206.jpeg?dl=0
Is it correct that way?

**Comment by bigtfromaz** [comment]
@deberlin 
That's what I wanted to see.   If you look at the output you can see that your container only has access to IPv4 addresses, e.g. 172.17.0.0/16, yet previous posts show that your host machine is IPv6 only.  Your container is isolated from everything except other containers that are in the same bridge network, no internet access.  

I think the next step is to connect your container to the host network and not the default bridge network.  

One way might be to stop the existing container, execute these commands and then restart the container.  I don't have a lot of experience with Docker networks so perhaps someone will add guidance if needed.

`sudo docker network disconnect bridge crashplan-pro`
`sudo docker network connect host crashplan-pro`

Another way, is if you use the GUI to create the container, delete the container, go to the image, recreate the container and under Advanced Settings and select "Use the same network as the Docker host".  Once the container is running we'll want to see ifconfig results from "inside" the container again.  

You could also recreate the container using docker run and use the `--net host`  parameter.

I know that @jlesage recommends bridge networks but IPv6 is a very different addressing scheme, and much safer.  If you have residential service from your ISP, your network prefix can and will change from time-to-time which will make it very difficult to keep the container connected to the CrashPlan service using a bridge. You will have to change the bridge prefix by hand each time your prefix changes.  If your ISP has assigned you a static IPv6 prefix then you could create a bridge but I really don't see the point for a stand-alone container like this one.

Let me know how it goes.

**Comment by beBerlin** [comment]
It's awesome.
I work.
Now I have only one more question.
How do I add folders from the DiskStation?
At HUI, only funny orders and files are sent to me.
I probably made a mistake during the installation.
I almost want to secure my complete DistStation at Crashplan.
Can I add the order ncoh later?

I thank you once again very much at all.

**Comment by bigtfromaz** [comment]
Are you starting with the GUI or a command?  Send the command you used to run the container for the first time, or screen snaps of your configuration screens.

**Comment by beBerlin** [comment]
Below the link you can see a screenshot. I have had these orders.

https://www.dropbox.com/s/3k1ohjm7895dfh7/Ohne%20Titel%205.jpeg?dl=0

**Comment by bigtfromaz** [comment]
It looks like you are mapping the host /volume1 to the container /volume1.  The container is configured to backup data stored in its /storage directory.  

So change `-v /volume1/:/volume1:ro` to `-v /volume1/:/storage:ro` .  This will make your Synology's volume1 visible to the container in /storage.

Let's see how that goes.

**Comment by beBerlin** [comment]
Can you tell me how to enter the commands without removing the containers and reinstalling?

**Comment by bigtfromaz** [comment]
No new installation needed.  No need to pull the image again.  Just remove the container.  The command to remove the container would be `docker container rm yourContainerName`  To list your containers use `docker container ls -a`  YourContainerName would be the --name value you supplied when creating the container with the docker run command.

There is no need to clear the config folder.  Change your docker run command as I suggested and run that command again.  Then check to see if the files show up.

**Comment by bigtfromaz** [comment]
@beBerlin It appears that you do not yet have a lot of experience with Linux, Docker, or CrashPlan Pro.  If you are doing this as a learning experience then keep reading and learning.  If you are doing this to manage a mission critical backup then I suggest that you consider switching to the GUI mode for creating and using this container on your Synology. The last thing you want is to find out your backup hasn't been running when you go to do a restore.  @jlesage has some good Synology notes in the readme for this container related to Synology.

I am pretty experienced in many technologies but chose to manage Docker on my Synology using the GUI.  I do this for convenience.  This way I have fewer scripts create and maintain.  @jlesage has created a very useful, well thought out and pretty complex image build. However, the Docker-run time requirements for this container are pretty basic, no swarms or complex firewalls required in Docker itself.  The GUI is convenient, reliable and meets the needs for a simple deployment like this one.

**Comment by beBerlin** [comment]
Yes, I have no idea.
That's why I ask for your help.
Unfortunately, it still does not work for me.
I do not understand how that can show the complete vo Volume1. For me are only funny folders that I know niche.
What information do you need to help me?
Maybe someone can help me with TeamViewer.
I almost finished the installation now. And do not want to give up now.

**Comment by bigtfromaz** [comment]
@beBerlin 

How many file shares do you have set up on your Synology?

**Comment by PNMarkW2** [comment]
Not an installation question as such, but I've been getting this message every time I've gone to check on my backup for the past couple of weeks.

"Routine maintenance. Backup will resume when maintenance completes."

Anyone else seeing this, or know it's meaning?

**Comment by jlesage** [comment]
It’s normally not an issue.  See https://support.code42.com/CrashPlan/6/Troubleshooting/Destination_unavailable, section “Under maintenance”.

**Comment by beBerlin** [comment]
I have 15 file shares.

**Comment by aagrawala** [comment]
Hi @excalibr18 & @jlesage, need some help... I initially did the setup successfully on Feb 4th this year. Everthing was working fine until about 26 days ago. So, somehow the backup is not running as reported by Crashplan in their emails to me and the web (see the attached snapshot).

<img width="646" alt="screen shot 2018-07-01 at 7 33 35 pm" src="https://user-images.githubusercontent.com/31644943/42142896-e6c590e0-7d66-11e8-919d-044540bd0f2c.png">

Do we need to download the image from the Registry section again to get the latest Crashplan? Or do something else? Do we have steps to be able to update Crashplan?

When I check the app by putting my IP address with 5800 port in my browser address field, I see that Crashplan is not able to update but the backup seems to have completed! (see the second snapshot)
<img width="836" alt="screen shot 2018-07-01 at 7 08 14 pm" src="https://user-images.githubusercontent.com/31644943/42142944-43d5fbc6-7d67-11e8-9be3-7be6ab8b2d23.png">


Thank you in advance!
Anil

**Comment by excalibr18** [comment]
@aagrawala, Crashplan pushed an update but the Docker image doesn't automatically apply it. You'll have to manually update the Docker image. Instructions for updating the Docker image can be found at:
https://github.com/jlesage/docker-crashplan-pro#synology

**Comment by PNMarkW2** [comment]
I need to change one of the environment variables. I did this once before, but it's been many months. My memory is I had to remove the container and recreate it with the new value. Is that correct? Seems rather drastic, but that's what I remember.

Thanks in advance.
Mark

**Comment by jlesage** [comment]
Yes that's the way to go.
Note that even if the container is re-created, data/state associated to CrashPlan is not lost (assuming it is re-created with the same parameters as before, with the exception of your environment variable).

**Comment by PNMarkW2** [comment]
Thanks. So I did that, but now I can't get to the web interface. I'm going to https://192.168.xxx.xxx:5800/ just like I've always done, but I get the message "This site can’t provide a secure connection" and "192.168.xxx.xxx sent an invalid response."

**Comment by jlesage** [comment]
Is the non-secure address working (http://192.168.xx.xx:5800)?

**Comment by PNMarkW2** [comment]
Yes, it is, I feel silly for not trying that. That's what I get for doing this right before I needed to head out so I was. Changing it to HTTP allowed me to see what's happening, and it now says it is synchronizing block information, which I assume is the correct state of affairs at this point. Like I said, it's been a long time since I last had to do this.

Is there a reason the https would not work when it would before? I was using a saved link to the web interface so I know it was the same link I used before I deleted the Docker container and recreated it with my new variable info.

**Comment by jlesage** [comment]
If HTTPs access was working before, it means that you probably forget to set the `SECURE_CONNECTION` to `1` when recreating the container?

**Comment by PNMarkW2** [comment]
Very possible, I could not find my original script for setting u the container.

Another issue that's come up, it finished the synchronizing block information, but it oddly thinks it's done and there is nothing to be backed up, which is not true, there is plenty to be backed up. Online I have the opposite, Code 42 is warning me there has been no activity in over 13 days (which is what prompted my needing to change the CRASHPLAN_SRV_MAX_MEM). Something does not seem to be synced between the container and online.

**Comment by jlesage** [comment]
In Crashplan, for your device, when you click Details->Manage Files, do you see your files?

**Comment by PNMarkW2** [comment]
That would be o. When I click on Manage Files I can see the folders that should contain files, but when I click into that folder there are no files visible.

**Comment by PNMarkW2** [comment]
Any thoughts on why my local install doesn't show any files, even though it does show my folders?

**Comment by gatorheel** [comment]
I had this when my new container arguments were wrong. It *looked* like I could see my folders, but it was really just showing me the structure from my prior backup.  I would double-check your -v paths to make sure you have them correct.

**Comment by jlesage** [comment]
Likely a permission issue.  Did you set the `USER_ID` and `GROUP_ID` to the same value as your original container?

**Comment by PNMarkW2** [comment]
Okay, I think between the two of you that you've helped highlight where I went wrong. I'm going to have to delete the container and recreate it to update some of the variables. I'll let you know the results.

**Comment by PNMarkW2** [comment]
On the plus side, I can see the full directory structure and the files on my network now. The downside seems to be that it's acting like it's starting over instead of resuming where it left off. I saw this because it claims to have backed up only 13GB of a multi-TB dataset. Now maybe it's only reporting what it's done since I restarted it, but it reads as if we're back at square one.

**Comment by PNMarkW2** [comment]
I tried again to delete the container and recreate it, only this time I read somewhere to remove any files from the system related to the old container. So yes, for good or for bad I did that. This time it started up much like when I created it the very first time by asking me to log in, but now it's stuck there. It just sits on that screen and says "Signing in..." and it's been that way for 7 hours now.

**Comment by jlesage** [comment]
Try to restart the container.
Also can you post the command line you used to create the container?

**Comment by PNMarkW2** [comment]
The Restart seemed to do some good, I had to log in again but I'm able to navigate around, it's not just stuck on "Signing in". However, it still looks like it's started over telling me that it's only 13% complete, that should be more like 50-60%.

As requested here is my startup script.
docker run -d \
--name=CrashPlan \
-e USER_ID=0 -e GROUP_ID=0 \
-e CRASHPLAN_SRV_MAX_MEM=3072M \
-e SECURE_CONNECTION=1 \
-p 5800:5800 \
-p 5900:5900 \
-v /volume1/docker/appdata/crashplan:/config:rw \
-v /volume1/:/volume1:ro \
--restart always \
jlesage/crashplan-pro

**Comment by jlesage** [comment]
Looking at the progression doesn't tell if your data is actually uploaded or deduplicated.  You can look at the history (Tools->History) for more details.

**Comment by jlesage** [comment]
You have too much `jlesage/crashplan-pro` in your command line.  It should be the last argument:
```
docker run -d --name=crashplan-pro -e USER_ID=0 -e GROUP_ID=0 -p 5800:5800 -v /volume1/docker/appdata/crashplan-pro:/config:rw -v /volume1/:/volume1:ro -v /volume2/:/volume2:ro -v /volume3/:/volume3:ro -v /volume4/:/volume4:ro -v /volume5/:/volume5:ro jlesage/crashplan-pro
```

**Comment by rekuhs** [comment]
> You have too much `jlesage/crashplan-pro` in your command line. It should be the last argument:
> 
> ```
> docker run -d --name=crashplan-pro -e USER_ID=0 -e GROUP_ID=0 -p 5800:5800 -v /volume1/docker/appdata/crashplan-pro:/config:rw -v /volume1/:/volume1:ro -v /volume2/:/volume2:ro -v /volume3/:/volume3:ro -v /volume4/:/volume4:ro -v /volume5/:/volume5:ro jlesage/crashplan-pro
> ```

Thank you :)

**Comment by THX-101** [comment]
I have a question about permissions: should I use -e USER_ID=0 -e GROUP_ID=0, or is this a security risk? Am running this on a Synology. Also am I or am I not supposed to run the "docker run -d ..." command as sudo?

**Comment by jlesage** [comment]
You are running as root when using  `USER_ID=0` and `GROUP_ID=0`, which is not considered as a good practice.  So if you are able, you should use a user/group that has permission to access the files you want to backup.

To create the container, you have the choice: either you manually run the `docker run` command, or you use the Synology UI.

**Comment by THX-101** [comment]
When using putty to build my containers, I log in with my Synology admin user, but I'm always required to run the `docker run` command with `sudo`. That is the way it is supposed to be, right?

And concerning not running as root, what would be best practice? Make a 'docker' group and a 'docker' user, that has read/write in both the /volume1/docker folder and /volume1/share and nothing else?

**Comment by jlesage** [comment]
Correct, the `docker run` command needs to be run as root, via `sudo`.

And yes, creating an additional user with restricted permissions is a solution.

**Comment by THX-101** [comment]
And one other question: I just ran the docker with `-e USER_ID=0 -e GROUP_ID=0` for the very first time (switched from Windows client). I suppose it is best to let Crashplan finish synchronising block information before I tear down the container, right?

**Comment by jlesage** [comment]
I think it should not be a problem to stop the container before it ends the synchronization.  Once you re-start the container, CrashPlan should just continue where it left off.

**Comment by THX-101** [comment]
Thanks jlesage, you are a one-man-tripple-A-customer support. Very much appreciated man 🥇

**Comment by PNMarkW2** [comment]
I recently had to reset my Synology from the ground up, which of course meant having to reset CrashPlan. So after I got the Synology running I added Docker followed closely by adding CrashPlan. Having had to reset CrashPlan once before I made sure to keep the setting I used.

docker run -d \
--name=CrashPlan \
-e USER_ID=0 -e GROUP_ID=0 \
-e CRASHPLAN_SRV_MAX_MEM=3072M \
-e SECURE_CONNECTION=1 \
-p 5800:5800 \
-p 5900:5900 \
-v /volume1/docker/appdata/crashplan:/config:rw \
-v /volume1/:/volume1:ro \
--restart always \
jlesage/crashplan-pro

But having done this all CrashPlan will do is constantly scan files. It goes for a while then restarts over and over. Today is 27 days since it performed any sort of backup. I've tried to delete it to reinstall, but when I try that I'm told that there are containers dependant on CrashPlan and it won't let me.

Any thoughts or suggestion would be welcome.

Thank you

Mark

**Comment by jlesage** [comment]
I would try to look at the history (View -> History) and at `/volume1/docker/appdata/crashplan/log/service.log` to see if there is anything obvious.
Also, I guess you are running the latest docker image?

**Comment by PNMarkW2** [comment]
Like taking a car to a mechanic, it spent a month scanning for files and seems to be "happy" now.

Under Tools -> History all it would appear to complain about it failing to upgrade to a newer version. As I said I had to re-do my entire Synology,  That involves downloading and installing Docker again, so I would assume it is the latest and greatest version available. I also had to get the CrashPlan package again, and again I would have assumed that to be the latest as well, but maybe not since it's trying to update so soon.

Right now it looks like it's backing up files, what still seems off though is it doesn't seem to think anything was backed up previously like it didn't sync with my previous backup after the reinstall. That doesn't quite match what I see when I log in to view my archive online, there it shows a good chunk of data with my last activity within the past 24 hours.

I'm still a bit confused, but it seems to be running, maybe. :-)

**Comment by jlesage** [comment]
For your information, I just published a new docker image containing the latest version of CP.

**Comment by PNMarkW2** [comment]
My CrashPlan installation on my Synology NAS has been running fine for months, actually nearly a year since my last post. Now I'm getting emails regularly that there hasn't been any backup, the email today said nothing has been backed up for 8 days.

So my first question is why would something that has been running fine for nearly a year just stop? And if it stopped because it's missing some update, why can't it tell me that instead of just stopping?

Second, when I try to bring up the local web interface I have to log in, which I do, and then I do again and again because it will sit there and say it's scanning files, then go to a black screen and sit there until asking me to log in again. This happens so quickly I'm not actually able to do anything when I get logged in, but I did get a look at the log which says there is an upgrade I need.

My guess is I have the nuke the whole thing and reinstall CrashPlan, but I wanted to check here first because that's not a process I enjoy.

Thank you for your help.

**Comment by jlesage** [comment]
CrashPlan will not perform any backup if its version is too old.  So if you didn't upgraded your container recently, then you need to do it to re-enable backup.

If you don't want to manually bother with upgrading your container, you could try to install Watchtower (https://github.com/containrrr/watchtower), a container which allows to seamlessly upgrade your containers.

**Comment by PNMarkW2** [comment]
Thank you for taking the time on a holiday to respond. I truly appreciate that, and I'll look into what you suggested since my efforts to follow the directions to update always seem to result in grief.

I will say again if the reason for the program to stop working is known, such as a version is too old, that perhaps that should be explicitly reported along with the email saying that the backup hasn't happened.

Thank you again.

Mark

**Comment by jlesage** [comment]
> I will say again if the reason for the program to stop working is known, such as a version is too old, that perhaps that should be explicitly reported along with the email saying that the backup hasn't happened.

You should forward this feedback to CrashPlan, since there is nothing I can do for that...

**Comment by SJLBoulder** [comment]
PNMarkW2 - I have a suggestion for your notification problem.  You can join GitHub and "watch" jlesage's project for updates, e.g. a new Docker container with the latest CrashPlan version.  Go here, and click on "watch" at the top of the page: https://github.com/jlesage/docker-crashplan-pro
